src/main/java/com/sequenceiq/cloudbreak/service/stack/resource/azure/builders/instance/AzureCloudServiceResourceBuilder.java

```diff
@@ -54,24 +54,12 @@
     private PollingService<AzureCloudServiceDeleteTaskContext> azureCloudServiceRemoveReadyPollerObjectPollingService;
 
     @Override
-    public List<Resource> create(AzureProvisionContextObject po, int index, List<Resource> resources) throws Exception {
-        Stack stack = stackRepository.findById(po.getStackId());
-        AzureTemplate azureTemplate = (AzureTemplate) stack.getTemplate();
-        AzureCredential azureCredential = (AzureCredential) stack.getCredential();
-        String vmName = getVmName(po.filterResourcesByType(ResourceType.AZURE_NETWORK).get(0).getResourceName(), index)
-                + String.valueOf(new Date().getTime());
-        if (vmName.length() > MAX_NAME_LENGTH) {
-            vmName = vmName.substring(vmName.length() - MAX_NAME_LENGTH, vmName.length());
-        }
-        Map<String, String> props = new HashMap<>();
-        props.put(NAME, vmName);
-        props.put(DESCRIPTION, azureTemplate.getDescription());
-        props.put(AFFINITYGROUP, po.getCommonName());
-        AzureClient azureClient = po.getNewAzureClient(azureCredential);
-        HttpResponseDecorator cloudServiceResponse = (HttpResponseDecorator) azureClient.createCloudService(props);
-        String requestId = (String) azureClient.getRequestId(cloudServiceResponse);
-        waitUntilComplete(azureClient, requestId);
-        return Arrays.asList(new Resource(ResourceType.AZURE_CLOUD_SERVICE, vmName, stack));
+    public Boolean create(final CreateResourceRequest cRR) throws Exception {
+        AzureCloudServiceCreateRequest aCSCR = (AzureCloudServiceCreateRequest) cRR;
```

Comment:
It can be argued but personally I don't like these names like aCSCR, cRR because they cannot be read easily. If you don't want to use azureCloudServiceCreateRequest here for example because its too long, you can simply use request or createRequest, it is understandable in the context.